craft_parts/overlays/layers.py

```diff
@@ -98,3 +101,51 @@ def save(self, part: Part) -> None:
     def hex(self) -> str:
         """Return the current hash value as a hexadecimal string."""
         return self.digest.hex()
+
+
+class LayerStateManager:
```

Comment:
maybe convert this into https://docs.python.org/3/library/collections.html#userdict-objects ?

And instead of passign parts, pass part_names, the index in compute_layer_hash can be derived from the part name.

Reply:
I was passing the part directory to `compute_layer_hash` to prevent some inefficiency in taking the name from a part and convert it back to a part. We can rebase it on UserDict, but I'd prefer to do it in a refactoring round after overlays are implemented in sequencer and executor so we can evaluate the interaction of all components in context.